### PR TITLE
Py3k: Use consistent sorting

### PR DESCRIPTION
**Pull request by QuLogic**
Iteration in Python 3 does not guarantee any particular order, so these should be sorted before asserting test results.

This PR is mostly concerned with test code, except for 254223eb42f78edd643dcc88d031f0ff4f2980d7 and 218635ea435169cdd3dd4d79484693dd73752431. The former is a minor change to an error message, and the latter is a little behavioural change to `load_files`, but I don't think it should be too troublesome.

### CODE REVIEW

**Comment by QuLogic** [comment]
There's one more sorting issue in `assertCML` that I haven't yet fixed. It's in `iris.tests.test_netcdf` which seems to produce cubes in a random order unlike other loaders. So should I sort the netcdf results or just sort during testing in `assertCML` (keep in mind that `Cube`s are no usually sortable.)

File: lib/iris/util.py [hunk 1]
```diff
@@ -613,8 +613,14 @@ def column_slices_generator(full_slice, ndims):
...
-    data_ndims = max(dimension_mapping.values())
-    if data_ndims is not None:
+    try:
+        data_ndims = max(dim
+                         for dim in dimension_mapping.values()
+                         if dim is not None)
+    except ValueError:
+        # No non-None dimensions
+        data_ndims = None
+    else:
```
Comment (rhattersley) [inline lib/iris/util.py]: How about we just ditch this code block? I think we can use `_count_current_dim` instead:
```python
spanning_slice_with_tuple = [slice(None, None)] * _count_current_dim
```

**Comment by rhattersley** [comment]
> should I sort the netcdf results or just sort during testing

How about sorting the cubes before calling `assertCML`? That's been [done before](https://github.com/SciTools/iris/blob/ee5918c9dbc3cd88713f1654849e3b1154e02c1d/lib/iris/tests/test_netcdf.py#L171).

**Comment by QuLogic** [comment]
Ah, okay, it looks like `cube.name()` is insufficient for some test data, but I've found that `cube.var_name` is unique.

**Comment by QuLogic** [comment]
And with that last commit, I'm now at 100% tests (except for skipped ESMF tests).

**Comment by rhattersley** [comment]
> And with that last commit, I'm now at 100% tests

:tada: :+1: :clap: :grinning:

File: lib/iris/util.py (hunk 1 above)
Comment (rhattersley) [inline lib/iris/util.py]: In case it helps I've sent you this as QuLogic/iris#1 ... feel free to ignore it if you've already done it.

**Pull request merged**